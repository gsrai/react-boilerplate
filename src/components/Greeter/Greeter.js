import React from 'react'
import PropTypes from 'prop-types'

const Greeter = ({ greeting = 'hello' }) => <span>{greeting}</span>

Greeter.propTypes = {
  greeting: PropTypes.string
}

export default Greeter
