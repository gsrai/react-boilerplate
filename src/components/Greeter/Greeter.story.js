import React from 'react'
import { storiesOf } from '@storybook/react'
import Greeter from './Greeter'

storiesOf('Components / Greeter', module)
  .add('default', () => <Greeter greeting='Hi Storybook' />)
