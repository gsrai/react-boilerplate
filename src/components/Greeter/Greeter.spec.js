import { shallow } from 'enzyme'
import React from 'react'
import Greeter from './Greeter'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(<Greeter />)
    expect(wrapper).toMatchSnapshot()
  })
})
