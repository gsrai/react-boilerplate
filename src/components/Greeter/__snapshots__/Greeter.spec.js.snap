// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<span>
  hello
</span>
`;
